Source/Core/DolphinQt/Debugger/CodeWidget.cpp

```diff
@@ -109,8 +111,16 @@ void CodeWidget::CreateWidgets()
   auto* top_layout = new QHBoxLayout;
   m_search_address = new QLineEdit;
   m_search_address->setPlaceholderText(tr("Search Address"));
+
+  m_lock_btn = new QToolButton();
+  m_lock_btn->setIcon(Resources::GetThemeIcon("pause"));
+  m_lock_btn->setCheckable(true);
+  m_lock_btn->setMinimumSize(24, 24);
+
```

Comment:
I don't think a simple icon is obvious and verbose enough for this feature to be understood. I think a menu item would better fit this feature as its item text would make it more explicit. For instance, with a checked by default item "Update address on pause" or similar.

However, I don't know if adding a menu with a single menu item to the code view is ideal either. I guess, the middle ground would be to provide a tool button with an appropriate icon _(ellipsis, settings gear, dolphin's config icon?, ...)_ and an associated menu that appears when clicked.

Otherwise, I suppose having a checkable menu item in the code view's right click menu is fine?

Reply:
<img width="550" height="107" alt="image" src="https://github.com/user-attachments/assets/94d8a8f1-8fd6-45af-b27a-fafdf014f644" />

Something like this?